tag_hotspots.py

```diff
@@ -24,79 +20,46 @@
-    sys.exit(1)
 
 def main():
-   parser = argparse.ArgumentParser(prog='tag_hotspots.py', description=' This tool helps to tag hotspot events', usage='%(prog)s [options]')
-   parser.add_argument("-v", "--verbose", action="store_true", dest="verbose", help="make lots of noise")
-   parser.add_argument("-m", "--input-maf", action="store", dest="inputMaf", required=True, type=str, metavar='SomeID.maf', help="Input maf file which needs to be tagged")
-   parser.add_argument("-itxt", "--input-hotspot", action="store", dest="inputTxt", required=True, type=str, metavar='SomeID.txt', help="Input txt file which has hotspots")
-   parser.add_argument("-o","--output-maf", action="store", dest="outputMaf", required=True, type=str, metavar='SomeID.maf', help="Output maf file name")
-   parser.add_argument("-outdir", "--outDir", action="store", dest="outdir", required=False, type=str, metavar='/somepath/output', help="Full Path to the output dir.")
-   
-   args = parser.parse_args()
-   if(args.verbose):
-       logger.info("tag_hotspots: Started the run for tagging hotspots")
-   (mafDF) =read_maf(args)
-   (hotspotDF) =read_hotspots(args)
-   (taggedDF) = tag_hotspots(args, mafDF, hotspotDF)
-   write_output(args,taggedDF)
-   if(args.verbose):
-       logger.info("tag_hotspots: Finished the run for tagging hotspots.")
+    parser = argparse.ArgumentParser(prog='tag_hotspots.py', description=' This tool helps to tag hotspot events', usage='%(prog)s [options]')
+    parser.add_argument("-v", "--verbose", action="store_true", dest="verbose", help="make lots of noise")
+    parser.add_argument("-m", "--input-maf", action="store", dest="inputMaf", required=True, type=str, metavar='SomeID.maf', help="Input maf file which needs to be tagged")
+    parser.add_argument("-itxt", "--input-hotspot", action="store", dest="inputTxt", required=True, type=str, metavar='SomeID.txt', help="Input txt file which has hotspots")
+    parser.add_argument("-o","--output-maf", action="store", dest="outputMaf", required=True, type=str, metavar='SomeID.maf', help="Output maf file name")
+    parser.add_argument("-outdir", "--outDir", action="store", dest="outdir", required=False, type=str, metavar='/somepath/output', help="Full Path to the output dir.")
 
-def read_maf(args):
-    dataDF = pd.read_table(args.inputMaf, comment="#", dtype=str)
-    return(dataDF)
+    args = parser.parse_args()
+    if(args.verbose):
+        logger.info("tag_hotspots: Started the run for tagging hotspots")
 
-def read_hotspots(args):
-    dataDF = pd.read_table(args.inputTxt, comment="#", dtype=str)
-    return(dataDF)
+    # Load hotspots into a dict for easy lookup by chr:pos:ref:alt, and store AA position changed
+    hotspot = dict()
+    with open(args.inputTxt, 'rb') as infile:
+        reader = csv.DictReader(infile, delimiter='\t')
+        for row in reader:
+            key = ':'.join([row['Chromosome'], row['Start_Position'], row['Reference_Allele'], row['Tumor_Seq_Allele2']])
```

Comment:
could it be extracted as a function to use in both places, here and while parsing maf?
If in the future key generation will change there will be only one place to change it instead of two

Reply:
Yea, I like functions for larger chunks of code. Another common key is "chrom:start:ref:alt:sample", if we want to differentiate the same variants in different samples. So there will be other cases where generating the key could get more complex and need its own function.